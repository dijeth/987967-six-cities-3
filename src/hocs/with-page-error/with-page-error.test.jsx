import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import withPageError from './with-page-error.jsx';
import NameSpace from '../../reducers/name-space.js';

const mockStore = configureStore([]);
const Component = () => <div>Mock</div>;
const ComponentWithPageError = withPageError(Component);

it(`should render error message when "pageError" === "Error"`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      pageError: `Error`
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ComponentWithPageError />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`should render Component when "pageError" === ""`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      pageError: ``
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <ComponentWithPageError />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
