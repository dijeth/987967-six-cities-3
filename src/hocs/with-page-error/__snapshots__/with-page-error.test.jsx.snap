// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Component when "pageError" === "" 1`] = `
<div>
  Mock
</div>
`;

exports[`should render error message when "pageError" === "Error" 1`] = `
Array [
  <div>
    Mock
  </div>,
  <div
    className="page page--gray page--login"
    style={
      Object {
        "position": "fixed",
        "top": 0,
        "width": "100vw",
        "zIndex": 1000,
      }
    }
  >
    <div
      className="page__main page__main--login"
    >
      <div
        className="page__login-container container"
      >
        <section
          className="login"
        >
          <h1
            className="login__title"
          >
            Error
          </h1>
          <button
            className="form__submit button"
            onClick={[Function]}
            type="button"
          >
            Продолжить
          </button>
        </section>
      </div>
    </div>
  </div>,
]
`;
